src/Core/src/Sts/StsClient.php

```diff
@@ -92,6 +93,15 @@ public function getCallerIdentity($input = []): GetCallerIdentityResponse
     protected function getEndpointMetadata(?string $region): array
     {
         switch ($region) {
+            case null:
```

Comment:
PHP 7.2+ optimizes switch statements when all cases are strings or integers (see https://derickrethans.nl/php7.2-switch.html). So I suggest keeping the case of `null` before the switch, so that the region switch is a jump table rather than a big if/elseif.

Also, due to using loose comparison, this actually allows the empty string to use the global endpoint.

Reply:
Thanks for the hint, you should reload the page, I fixed it yesterday